Minifier/rewriter.fs

```diff
@@ -469,17 +469,30 @@ type private RewriterImpl(options: Options.Options, optimizationPass: Optimizati
 
         List.collect replacements stmts
 
+    // Helper for deterimining if all array dimensions in 2 
+    // declaration lists are equal.
+    let all_sizes_equal li1 li2 =
+        match li1 @ li2 with
+        | [] -> true
+        | first :: rest ->
+            let sizes = first.sizes
+            List.forall (fun decl -> decl.sizes = sizes) rest
```

Comment:
I'm not sure I understand this implementation.
Shouldn't [4,3] and [4,3] be considered equal?